src/jaxsim/mujoco/loaders.py

```diff
@@ -0,0 +1,480 @@
+            balanceinertia="true",
+            discardvisual="false",
+        )
+
+        urdf_string = ET.tostring(root, pretty_print=True).decode()
+        # print(urdf_string)
+        # raise
+
+        # ------------------------------
+        # Post-process all dummy visuals
+        # ------------------------------
+
+        parser = ET.XMLParser(remove_blank_text=True)
+        root: ET._Element = ET.fromstring(text=urdf_string.encode(), parser=parser)
+        import numpy as np
+
+        # Give a tiny radius to all dummy spheres
+        for geometry in root.findall(".//visual/geometry[sphere]"):
+            radius = np.fromstring(
+                geometry.find("./sphere").attrib["radius"], sep=" ", dtype=float
+            )
+            if np.allclose(radius, np.zeros(1)):
+                geometry.find("./sphere").set("radius", "0.001")
+
+        # Give a tiny volume to all dummy boxes
+        for geometry in root.findall(".//visual/geometry[box]"):
+            size = np.fromstring(
+                geometry.find("./box").attrib["size"], sep=" ", dtype=float
+            )
+            if np.allclose(size, np.zeros(3)):
+                geometry.find("./box").set("size", "0.001 0.001 0.001")
+
+        urdf_string = ET.tostring(root, pretty_print=True).decode()
+
+        # ------------------------
+        # Convert the URDF to MJCF
+        # ------------------------
+
+        # Load the URDF model into Mujoco.
+        assets = RodModelToMjcf.assets_from_rod_model(rod_model=rod_model)
+        mj_model = mj.MjModel.from_xml_string(xml=urdf_string, assets=assets)  # noqa
+
+        # Get the joint names.
+        mj_joint_names = set(
+            [
+                mj.mj_id2name(mj_model, mj.mjtObj.mjOBJ_JOINT, idx)
+                for idx in range(mj_model.njnt)
+            ]
+        )
+
+        # Check that the Mujoco model only has the considered joints.
+        if mj_joint_names != considered_joints:
+            extra1 = mj_joint_names - considered_joints
+            extra2 = considered_joints - mj_joint_names
+            extra_joints = extra1.union(extra2)
+            msg = "The Mujoco model has the following extra/missing joints: '{}'"
+            raise ValueError(msg.format(extra_joints))
+
+        with tempfile.NamedTemporaryFile(
+            mode="w+", suffix=".xml", prefix=f"{rod_model.name}_"
+        ) as mjcf_file:
+
+            # Convert the in-memory Mujoco model to MJCF.
+            mj.mj_saveLastXML(mjcf_file.name, mj_model)
+
+            # Parse the MJCF string as XML (etree).
+            # We need to post-process the file to include additional elements.
+            parser = ET.XMLParser(remove_blank_text=True)
+            tree = ET.parse(source=mjcf_file, parser=parser)
+
+        # Get the root element.
+        root: ET._Element = tree.getroot()
+
+        # Find the <mujoco> element (might be the root itself).
+        mujoco_elements = [e for e in root.iter("mujoco")]
```

Comment:
```suggestion
        mujoco_elements = list(root.iter("mujoco"))
```
The list comprehension shouldn't be necessary here

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/83/commits/4dcd4a760ac2d9afe7d6a74fea385f2447ebb07f